core/data/data.go

```diff
@@ -32,6 +37,66 @@ func SetupPersistence() {
 		}
 	}
 
-	sqliteDatabase, _ := sql.Open("sqlite3", file)
-	_db = sqliteDatabase
+	db, err := sql.Open("sqlite3", file)
+	if err != nil {
+		return err
+	}
+
+	if _, err := db.Exec(`CREATE TABLE IF NOT EXISTS config (
+		"key" string NOT NULL PRIMARY KEY,
+		"value" TEXT
+	);`); err != nil {
+		return err
+	}
+
+	var version int
+	err = db.QueryRow("SELECT value FROM config WHERE key='version'").
+		Scan(&version)
+	if err != nil {
+		if err != sql.ErrNoRows {
+			return err
+		}
+
+		// fresh database: initialize it with the current schema version
+		_, err := db.Exec("INSERT INTO config(key, value) VALUES(?, ?)", "version", schemaVersion)
+		if err != nil {
+			return err
+		}
+		version = schemaVersion
+	}
+
+	// is database from a newer Owncast version?
+	if version > schemaVersion {
+		return fmt.Errorf("incompatible database version %d (versions up to %d are supported)",
```

Comment:
For consistency could you use `log` instead of `fmt`?

Reply:
This isn't printing to screen though, this is merely generating an `error`. The error gets caught by `main()` and will use `log` to print it to screen.

Reply:
Sorry, I commented on the wrong line:

86: fmt.Printf("Migrating database from version %d to %d\n", from, to)
90: fmt.Printf("Migration step from %d to %d\n", v, v+1)

Reply:
Done!